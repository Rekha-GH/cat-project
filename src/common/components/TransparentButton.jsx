import React from "react";
import { classNames } from '../Utils/classUtils';

const DEFAULT_CLASSNAME = "transparent-button";

const TransparentButton = React.memo((props) => {
  const resultClass = classNames(DEFAULT_CLASSNAME, props.classes);

  return (
    <button type={"button"} className={resultClass} onClick={props.action}>
      {props.children}
    </button>
  );
});

export default TransparentButton;
