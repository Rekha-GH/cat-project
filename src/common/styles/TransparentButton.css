.transparent-button {
  background: transparent;
  border: transparent;
  color: #fff;
  font-size: 21px;
}
