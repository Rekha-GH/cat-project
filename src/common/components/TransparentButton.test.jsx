import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import TransparentButton from "./TransparentButton";

Enzyme.configure({ adapter: new Adapter() });

describe("TransparentButton Test", () => {
  test("TransparentButton Call Correctly", () => {
    const wrapper = shallow(<TransparentButton>test</TransparentButton>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("TransparentButton triggers click event", () => {
    let flg = false;
    const wrapper = shallow(
      <TransparentButton
        action={() => {
          flg = true;
        }}
      >
        test
      </TransparentButton>
    );

    wrapper.find("button").simulate("click");

    expect(flg).toBe(true);
  });
});
