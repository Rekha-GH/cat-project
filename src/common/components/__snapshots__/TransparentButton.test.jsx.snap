// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransparentButton Test TransparentButton Call Correctly 1`] = `
<button
  className="transparent-button "
  type="button"
>
  test
</button>
`;
